--- vsxu-0.4.2/artiste/vsxu-artiste.desktop.in.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/artiste/vsxu-artiste.desktop.in	2014-01-04 21:48:33.094109544 +0100
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Name=VSXu Artiste [windowed]
+Name[pl]=VSXu Artiste [w oknie]
 GenericName=Real Time OpenGL Graphics Editor
 Comment=VSXu Artiste - Create Real Time Graphics
-Exec=/usr/bin/env LD_LIBRARY_PATH=@CMAKE_INSTALL_PREFIX@/lib @CMAKE_INSTALL_PREFIX@/bin/vsxu_artiste
+Comment[pl]=VSXu Artiste - tworzenie grafiki w czasie rzeczywistym
+Exec=@CMAKE_INSTALL_PREFIX@/bin/vsxu_artiste
 Icon=vsxu
 Type=Application
 Categories=Graphics;
--- vsxu-0.4.2/artiste/vsxu-artiste-fullscreen.desktop.in.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/artiste/vsxu-artiste-fullscreen.desktop.in	2014-01-04 21:49:23.687440702 +0100
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Name=VSXu Artiste [fullscreen]
+Name[pl]=VSXu Artiste [pełny ekran]
 GenericName=Real Time OpenGL Graphics Editor
 Comment=VSXu Artiste - Create Real Time Graphics
-Exec=/usr/bin/env LD_LIBRARY_PATH=@CMAKE_INSTALL_PREFIX@/lib @CMAKE_INSTALL_PREFIX@/bin/vsxu_artiste -f
+Comment[pl]=VSXu Artiste - tworzenie grafiki w czasie rzeczywistym
+Exec=@CMAKE_INSTALL_PREFIX@/bin/vsxu_artiste -f
 Icon=vsxu
 Type=Application
 Categories=Graphics;
--- vsxu-0.4.2/player/vsxu-player.desktop.in.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/player/vsxu-player.desktop.in	2014-01-04 21:50:22.114104932 +0100
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=VSXu Player [windowed]
+Name[pl]=VSXu Player [w oknie]
 GenericName=Music Visualizer
 Comment=VSXu Player
-Exec=/usr/bin/env LD_LIBRARY_PATH=@CMAKE_INSTALL_PREFIX@/lib @CMAKE_INSTALL_PREFIX@/bin/vsxu_player
+Exec=@CMAKE_INSTALL_PREFIX@/bin/vsxu_player
 Icon=vsxu
 Type=Application
 Categories=AudioVideo;Audio;Player;
--- vsxu-0.4.2/player/vsxu-player-fullscreen.desktop.in.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/player/vsxu-player-fullscreen.desktop.in	2014-01-04 21:50:43.957437407 +0100
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=VSXu Player [fullscreen]
+Name[pl]=VSXu Player [pełny ekran]
 GenericName=Music Visualizer
 Comment=VSXu Player
-Exec=/usr/bin/env LD_LIBRARY_PATH=@CMAKE_INSTALL_PREFIX@/lib @CMAKE_INSTALL_PREFIX@/bin/vsxu_player -f
+Exec=@CMAKE_INSTALL_PREFIX@/bin/vsxu_player -f
 Icon=vsxu
 Type=Application
 Categories=AudioVideo;Audio;Player;
--- vsxu-0.4.2/server/vsxu-server.desktop.in.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/server/vsxu-server.desktop.in	2014-01-04 21:51:22.134102443 +0100
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Name=VSXu server [windowed]
+Name[pl]=VSXu server [w oknie]
 GenericName=Real Time OpenGL Graphics Server
 Comment=VSXu Server - Perform Real Time Graphics
-Exec=/usr/bin/env LD_LIBRARY_PATH=@CMAKE_INSTALL_PREFIX@/lib @CMAKE_INSTALL_PREFIX@/bin/vsxu_server
+Comment[pl]=VSXu Server - wykonywanie grafiki w czasie rzeczywistym
+Exec=@CMAKE_INSTALL_PREFIX@/bin/vsxu_server
 Icon=vsxu
 Type=Application
 Categories=Graphics;
--- vsxu-0.4.2/server/vsxu-server-fullscreen.desktop.in.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/server/vsxu-server-fullscreen.desktop.in	2014-01-04 21:51:53.657434467 +0100
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Name=VSXu server [fullscreen]
+Name[pl]=VSXu server [pełny ekran]
 GenericName=Real Time OpenGL Graphics Server
 Comment=VSXu Server - Perform Real Time Graphics
-Exec=/usr/bin/env LD_LIBRARY_PATH=@CMAKE_INSTALL_PREFIX@/lib @CMAKE_INSTALL_PREFIX@/bin/vsxu_server -f
+Comment[pl]=VSXu Server - wykonywanie grafiki w czasie rzeczywistym
+Exec=@CMAKE_INSTALL_PREFIX@/bin/vsxu_server -f
 Icon=vsxu
 Type=Application
 Categories=Graphics;
