--- vsxu-0.6.3/cmake/modules/FindGLFW.cmake.orig	2020-03-20 06:19:07.835811451 +0100
+++ vsxu-0.6.3/cmake/modules/FindGLFW.cmake	2020-03-20 06:22:59.794554823 +0100
@@ -20,7 +20,7 @@
 # Allow the user to select to link to a shared library or to a static library.
 
 #Search for the include file...
-FIND_PATH(GLFW_INCLUDE_PATH GL/glfw.h DOC "Path to GLFW include directory."
+FIND_PATH(GLFW_INCLUDE_PATH GLFW/glfw3.h DOC "Path to GLFW include directory."
   PATHS
     $ENV{DEPINSTALL}/include
     /usr/include/
--- vsxu-0.6.3/lib/application/src/glfw/vsx_mouse.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.6.3/lib/application/src/glfw/vsx_mouse.cpp	2016-03-02 18:33:29.197403578 +0100
@@ -22,7 +22,7 @@
 */
 
 #include "vsx_mouse.h"
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 
 void vsx_mouse::set_cursor(int id)
 {
@@ -37,7 +37,7 @@
 {
   // This causes problems with knobs & arcballs, mouse visibility is disabled for now
   //glfwEnable(GLFW_MOUSE_CURSOR);
-  //glfwSetMousePos((int)(position.x), (int)(position.y));
+  //glfwSetCursorPos(glfwGetCurrentContext(), (int)(position.x), (int)(position.y));
 }
 
 void vsx_mouse::hide_cursor()
@@ -50,7 +50,7 @@
 void vsx_mouse::set_cursor_pos(float x, float y)
 {
   position.set(x,y);
-  glfwSetMousePos((int)(position.x), (int)(position.y));
+  glfwSetCursorPos(glfwGetCurrentContext(), (int)(position.x), (int)(position.y));
 }
 
 vsx_mouse::vsx_mouse()
