--- vsxu-0.4.2/cmake/modules/FindGLFW.cmake.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/cmake/modules/FindGLFW.cmake	2014-01-01 17:59:46.528896515 +0100
@@ -20,7 +20,7 @@
 # Allow the user to select to link to a shared library or to a static library.
 
 #Search for the include file...
-FIND_PATH(GLFW_INCLUDE_PATH GL/glfw.h DOC "Path to GLFW include directory."
+FIND_PATH(GLFW_INCLUDE_PATH GLFW/glfw3.h DOC "Path to GLFW include directory."
   HINTS
   $ENV{GLFW_ROOT}
   PATH_SUFFIX include #For finding the include file under the root of the glfw expanded archive, typically on Windows.
--- vsxu-0.5.1/artiste/src/vsx_widget/server/vsx_widget_comp.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/artiste/src/vsx_widget/server/vsx_widget_comp.cpp	2016-03-02 17:54:59.490833843 +0100
@@ -45,7 +45,7 @@
 #include "widgets/vsx_widget_popup_menu.h"
 #include "vsx_data_path.h"
 #include <gl_helper.h>
-#include <GL/glfw.h>
+#include <GLFW/glfw3.h>
 
 // widget
 #include <dialogs/dialog_query_string.h>
@@ -1680,7 +1680,7 @@
   VSX_UNUSED(alt);
   VSX_UNUSED(ctrl);
   VSX_UNUSED(shift);
-  if (abs(key) == GLFW_KEY_DEL)
+  if (abs(key) == GLFW_KEY_DELETE)
   {
     if (((vsx_widget_server*)(this->server))->selected_list.size())
     for (std::list<vsx_widget*>::iterator itx = ((vsx_widget_server*)server)->selected_list.begin(); itx != ((vsx_widget_server*)server)->selected_list.end(); ++itx) {
--- vsxu-0.5.1/artiste/src/artiste_main.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/artiste/src/artiste_main.cpp	2016-03-02 18:31:13.680742600 +0100
@@ -27,7 +27,7 @@
 #include <vsx_string_helper.h>
 #include <vsx_argvector.h>
 #include <GL/glew.h>
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 #include "artiste_application.h"
 #include "vsxfst.h"
 #include "vsx_version.h"
@@ -70,9 +70,9 @@
 
 void set_modifiers()
 {
-  app_ctrl = (bool)glfwGetKey(GLFW_KEY_LCTRL);
-  app_shift = (bool)glfwGetKey(GLFW_KEY_LSHIFT);
-  app_alt = (bool)glfwGetKey(GLFW_KEY_LALT);
+  app_ctrl = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_CONTROL);
+  app_shift = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_SHIFT);
+  app_alt = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_ALT);
 }
 
 long key_pressed = -1;
@@ -81,14 +81,10 @@
 float key_repeat_time;
 float initial_key_delay = 0.04f;
 
-void GLFWCALL key_char_event( int character, int action )
+void key_char_event(GLFWwindow *window, unsigned int character)
 {
-  if (action == GLFW_PRESS)
-  {
     app_char(character);
     key_character = character;
-
-  }
 }
 
 void app_close_window()
@@ -93,10 +89,10 @@
 
 void app_close_window()
 {
-  glfwCloseWindow();
+  glfwDestroyWindow(glfwGetCurrentContext());
 }
 
-void GLFWCALL key_event(int key, int action)
+void key_event(GLFWwindow *window, int key, int scancode, int action, int mods)
 {
   set_modifiers();
   if (action == GLFW_PRESS)
@@ -115,12 +111,12 @@
   }
 }
 
-int last_x = 0, last_y = 0;
+double last_x = 0, last_y = 0;
 int mouse_state = 0;
 
-void GLFWCALL mouse_button_event(int button, int action)
+void mouse_button_event(GLFWwindow *window, int button, int action, int mods)
 {
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   set_modifiers();
   unsigned long i_but = 0;
   switch (button)
@@ -142,26 +138,26 @@
 
 int mouse_pos_type = 0;
 
-void GLFWCALL mouse_pos_event(int x, int y)
+void mouse_pos_event(GLFWwindow *window, double x, double y)
 {
   VSX_UNUSED(x);
   VSX_UNUSED(y);
   set_modifiers();
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   if (mouse_state) mouse_pos_type = 1;
   else mouse_pos_type = 2;
 }
 
-int mousewheel_prev_pos = 0;
+double mousewheel_prev_pos = 0;
 
-void GLFWCALL mouse_wheel(int pos)
+void mouse_wheel(GLFWwindow *window, double xoffset, double yoffset)
 {
   set_modifiers();
-  app_mousewheel((float)(pos-mousewheel_prev_pos),last_x,last_y);
-  mousewheel_prev_pos = pos;
+  app_mousewheel((float)(yoffset-mousewheel_prev_pos),last_x,last_y);
+  mousewheel_prev_pos = yoffset;
 }
 
-void GLFWCALL window_size( int width, int height )
+void window_size( GLFWwindow *window, int width, int height )
 {
   vsx_gl_state::get_instance()->viewport_change(0,0,width, height);
 }
@@ -206,7 +202,8 @@
   glfwInit();
   set_modifiers();
 
-  int     width, height, running, frames, x, y;
+  int     width, height, running, frames;
+  double  x, y;
   double  t, t1;
   char    titlestr[ 200 ];
 
@@ -249,24 +246,24 @@
   if (start_fullscreen && !manual_resolution_set)
   {
     // try to get the resolution from the desktop for fullscreen
-    GLFWvidmode video_mode;
-    glfwGetDesktopMode(&video_mode);
-    x_res = video_mode.Height;
-    y_res = video_mode.Width;
+    GLFWvidmode video_mode = * glfwGetVideoMode(glfwGetPrimaryMonitor());
+    x_res = video_mode.height;
+    y_res = video_mode.width;
   }
 
 
   if (vsx_argvector::get_instance()->has_param("gl_debug"))
   {
     printf("enabling GL DEBUG\n");
-    glfwOpenWindowHint( GLFW_OPENGL_DEBUG_CONTEXT , GL_TRUE );
+    glfwWindowHint( GLFW_OPENGL_DEBUG_CONTEXT , GL_TRUE );
   }
 
   // OpenGL version
-  glfwOpenWindowHint(GLFW_OPENGL_VERSION_MAJOR, 2);
-  glfwOpenWindowHint(GLFW_OPENGL_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_DEPTH_BITS, 16);
 
-  if( !glfwOpenWindow( x_res, y_res, 0,0,0,0,16,0, start_fullscreen?GLFW_FULLSCREEN:GLFW_WINDOW ) ) // GLFW_FULLSCREEN
+  if( !glfwCreateWindow( x_res, y_res, "", start_fullscreen ? glfwGetPrimaryMonitor() : NULL, NULL ) )
   {
     printf("Error! Could not create an OpenGL context. Please check your GPU drivers...\n");
     glfwTerminate();
@@ -277,12 +274,10 @@
 
 
   if (start_fullscreen)
-    glfwEnable( GLFW_MOUSE_CURSOR );
+    glfwSetInputMode(glfwGetCurrentContext(), GLFW_CURSOR, GLFW_CURSOR_NORMAL);
 
   app_load(0);
 
-  glfwEnable(GLFW_AUTO_POLL_EVENTS);
-
   for (int i = 1; i < argc; i++)
   {
     vsx_string arg1 = argv[i];
@@ -294,20 +289,20 @@
         vsx_avector<vsx_string> parts;
         vsx_string deli = ",";
         explode(arg2, deli, parts);
-        glfwSetWindowPos( vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]) );
+        glfwSetWindowPos(glfwGetCurrentContext(), vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]));
       }
     }
   }
 
-  glfwSetKeyCallback(&key_event);
-  glfwSetMouseButtonCallback(&mouse_button_event);
-  glfwSetMousePosCallback(&mouse_pos_event);
-  glfwSetCharCallback(&key_char_event);
-  glfwSetMouseWheelCallback(&mouse_wheel);
+  glfwSetKeyCallback(glfwGetCurrentContext(), &key_event);
+  glfwSetMouseButtonCallback(glfwGetCurrentContext(), &mouse_button_event);
+  glfwSetCursorPosCallback(glfwGetCurrentContext(), &mouse_pos_event);
+  glfwSetCharCallback(glfwGetCurrentContext(), &key_char_event);
+  glfwSetScrollCallback(glfwGetCurrentContext(), &mouse_wheel);
   // set window size callback function
-  glfwSetWindowSizeCallback(window_size);
+  glfwSetWindowSizeCallback(glfwGetCurrentContext(), &window_size);
   // Enable sticky keys
-  glfwEnable( GLFW_STICKY_KEYS );
+  glfwSetInputMode(glfwGetCurrentContext(), GLFW_STICKY_KEYS, GL_TRUE);
 
 
   // vsync handling
@@ -356,7 +351,7 @@
   #if PLATFORM_FAMILY == PLATFORM_FAMILY_WINDOWS
   sprintf( titlestr, "Vovoid VSXu Artiste %s [Windows %d-bit]", vsxu_ver, PLATFORM_BITS);
   #endif
-  glfwSetWindowTitle( titlestr );
+  glfwSetWindowTitle( glfwGetCurrentContext(), titlestr );
 
 
   vsx_timer frame_delay;
@@ -409,7 +404,7 @@
 
     // Get time and mouse position
     t = glfwGetTime();
-    glfwGetMousePos( &x, &y );
+    glfwGetCursorPos(glfwGetCurrentContext(), &x, &y );
     float delta = t-t1;
     t1 = t;
     if (key_pressed != -1)
@@ -435,7 +430,7 @@
     frames ++;
 
       // Get window size (may be different than the requested size)
-      glfwGetWindowSize( &width, &height );
+      glfwGetWindowSize( glfwGetCurrentContext(), &width, &height );
       height = height > 0 ? height : 1;
 
       // Set viewport
@@ -451,7 +446,7 @@
 
     app_draw(0);
 
-    glfwSwapBuffers();
+    glfwSwapBuffers(glfwGetCurrentContext());
 
 //#if (PLATFORM != PLATFORM_WINDOWS)
 //    if (!vsync)
@@ -467,7 +462,7 @@
 //#endif
 
 
-    running = glfwGetWindowParam( GLFW_OPENED );
+    running = !glfwWindowShouldClose(glfwGetCurrentContext());
   }
 
   app_unload();
--- vsxu-0.5.1/player/src/main.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/player/src/main.cpp	2016-03-02 18:37:43.074059591 +0100
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include "application.h"
 #include <GL/glew.h>
-#include <GL/glfw.h>
+#include <GLFW/glfw3.h>
 #include <container/vsx_avector.h>
 #include <vsx_string.h>
 #include <vsx_string_helper.h>
@@ -48,9 +48,9 @@
 
 void set_modifiers()
 {
-  app_ctrl = (bool)glfwGetKey(GLFW_KEY_LCTRL);
-  app_shift = (bool)glfwGetKey(GLFW_KEY_LSHIFT);
-  app_alt = (bool)glfwGetKey(GLFW_KEY_LALT);
+  app_ctrl = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_CONTROL);
+  app_shift = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_SHIFT);
+  app_alt = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_ALT);
 }
 
 long key_pressed = -1;
@@ -59,18 +59,15 @@
 float key_repeat_time;
 float initial_key_delay = 0.04f;
 
-void GLFWCALL key_char_event( int character, int action )
+void key_char_event( GLFWwindow *window, unsigned int character )
 {
-  if (action == GLFW_PRESS)
-  {
     //printf("got key: %d\n",character);
     app_char(character);
     key_character = character;
-  }
 }
 
 
-void GLFWCALL key_event(int key, int action)
+void key_event(GLFWwindow *window, int key, int scancode, int action, int mods)
 {
   set_modifiers();
   if (action == GLFW_PRESS)
@@ -91,12 +88,12 @@
   }
 }
 
-int last_x = 0, last_y = 0;
+double last_x = 0, last_y = 0;
 int mouse_state = 0;
 
-void GLFWCALL mouse_button_event(int button, int action)
+void mouse_button_event(GLFWwindow *window, int button, int action, int mods)
 {
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   set_modifiers();
   unsigned long i_but = 0;
   switch (button)
@@ -118,25 +115,25 @@
 
 int mouse_pos_type = 0;
 
-void GLFWCALL mouse_pos_event(int x, int y)
+void mouse_pos_event(GLFWwindow *window, double x, double y)
 {
   VSX_UNUSED(x);
   VSX_UNUSED(y);
   set_modifiers();
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   if (mouse_state) mouse_pos_type = 1;
   else mouse_pos_type = 2;
 }
 
-int mousewheel_prev_pos = 0;
+double mousewheel_prev_pos = 0;
 
-void GLFWCALL mouse_wheel(int pos)
+void mouse_wheel(GLFWwindow *window, double xoffset, double yoffset)
 {
-  app_mousewheel((float)(pos-mousewheel_prev_pos),last_x,last_y);
-  mousewheel_prev_pos = pos;
+  app_mousewheel((float)(yoffset-mousewheel_prev_pos),last_x,last_y);
+  mousewheel_prev_pos = yoffset;
 }
 
-void GLFWCALL window_size( int width, int height )
+void window_size( GLFWwindow *window, int width, int height )
 {
   vsx_gl_state::get_instance()->viewport_change(0,0,width, height);
 }
@@ -149,7 +146,8 @@
 {
   app_argc = argc;
   app_argv = argv;
-  int     width, height, running, frames, x, y;
+  int     width, height, running, frames;
+  double  x, y;
   double  t, t1;
   char    titlestr[ 200 ];
 
@@ -220,15 +218,15 @@
   if (start_fullscreen && !manual_resolution_set)
   {
     // try to get the resolution from the desktop for fullscreen
-    GLFWvidmode video_mode;
-    glfwGetDesktopMode(&video_mode);
-    x_res = video_mode.Height;
-    y_res = video_mode.Width;
+    GLFWvidmode video_mode = * glfwGetVideoMode(glfwGetPrimaryMonitor());
+    x_res = video_mode.height;
+    y_res = video_mode.width;
   }
   
   // Open OpenGL window
-  glfwOpenWindowHint(GLFW_FSAA_SAMPLES, 4);
-  if( !glfwOpenWindow( x_res, y_res, 0,0,0,0,16,0, start_fullscreen?GLFW_FULLSCREEN:GLFW_WINDOW ) ) // GLFW_FULLSCREEN
+  glfwWindowHint(GLFW_SAMPLES, 4);
+  glfwWindowHint(GLFW_DEPTH_BITS, 16);
+  if( !glfwCreateWindow( x_res, y_res, "", start_fullscreen ? glfwGetPrimaryMonitor() : NULL, NULL ) )
   {
     printf("Error! Could not create an OpenGL context. Please check your GPU drivers...\n");
     glfwTerminate();
@@ -237,14 +235,12 @@
 
   glewInit();
 
-  if (start_fullscreen) glfwEnable( GLFW_MOUSE_CURSOR );
+  if (start_fullscreen) glfwSetInputMode(glfwGetCurrentContext(), GLFW_CURSOR, GLFW_CURSOR_NORMAL);
 
   printf("INFO: app_init\n");
   app_init(0);
   printf("INFO: app_init done\n");
 
-  glfwEnable(GLFW_AUTO_POLL_EVENTS);
-
   for (int i = 1; i < argc; i++) {
     vsx_string arg1 = argv[i];
     if (arg1 == "-p") {
@@ -255,21 +251,21 @@
         vsx_avector<vsx_string> parts;
         vsx_string deli = ",";
         explode(arg2, deli, parts);
-        glfwSetWindowPos( vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]) );
+        glfwSetWindowPos(glfwGetCurrentContext(), vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]));
       }
     }
   }
 
-  glfwSetKeyCallback(&key_event);
-  glfwSetMouseButtonCallback(&mouse_button_event);
-  glfwSetMousePosCallback(&mouse_pos_event);
-  glfwSetCharCallback(&key_char_event);
-  glfwSetMouseWheelCallback(&mouse_wheel);
+  glfwSetKeyCallback(glfwGetCurrentContext(), &key_event);
+  glfwSetMouseButtonCallback(glfwGetCurrentContext(), &mouse_button_event);
+  glfwSetCursorPosCallback(glfwGetCurrentContext(), &mouse_pos_event);
+  glfwSetCharCallback(glfwGetCurrentContext(), &key_char_event);
+  glfwSetScrollCallback(glfwGetCurrentContext(), &mouse_wheel);
   // set window size callback function
-  glfwSetWindowSizeCallback(window_size);
+  glfwSetWindowSizeCallback(glfwGetCurrentContext(), &window_size);
 
   // Enable sticky keys
-  glfwEnable( GLFW_STICKY_KEYS );
+  glfwSetInputMode(glfwGetCurrentContext(), GLFW_STICKY_KEYS, GL_TRUE);
   glfwSwapInterval(1);
 
   // Main loop
@@ -282,7 +278,7 @@
   #if PLATFORM_FAMILY == PLATFORM_FAMILY_WINDOWS
     sprintf( titlestr, "Vovoid VSXu Player %s [Windows %d-bit]", vsxu_ver, PLATFORM_BITS);
   #endif
-  glfwSetWindowTitle( titlestr );
+  glfwSetWindowTitle( glfwGetCurrentContext(), titlestr );
 
 
   while( running )
@@ -299,7 +295,7 @@
 
     // Get time and mouse position
     t = glfwGetTime();
-    glfwGetMousePos( &x, &y );
+    glfwGetCursorPos( glfwGetCurrentContext(), &x, &y );
     float delta = t-t1;
     t1 = t;
     if (key_pressed != -1)
@@ -323,7 +319,7 @@
     frames ++;
 
     // Get window size (may be different than the requested size)
-    glfwGetWindowSize( &width, &height );
+    glfwGetWindowSize( glfwGetCurrentContext(), &width, &height );
     height = height > 0 ? height : 1;
 
     // Set viewport
@@ -339,11 +335,11 @@
 
     app_draw(0);
 
-    glfwSwapBuffers();
+    glfwSwapBuffers(glfwGetCurrentContext());
 
     // Check if the ESC key was pressed or the window was closed
-    running = /*!glfwGetKey( GLFW_KEY_ESC ) &&*/
-    glfwGetWindowParam( GLFW_OPENED );
+    running = /*!glfwGetKey( glfwGetCurrentContext(), GLFW_KEY_ESCAPE ) &&*/
+    	!glfwWindowShouldClose( glfwGetCurrentContext() );
   }
 
   // Close OpenGL window and terminate GLFW
--- vsxu-0.4.2/player/src/application.cpp.orig	2013-11-24 11:00:51.000000000 +0100
+++ vsxu-0.4.2/player/src/application.cpp	2014-01-02 19:53:33.364984154 +0100
@@ -36,7 +36,7 @@
 // you might want to remove this.
 #include "vsx_overlay.h"
 
-#include <GL/glfw.h>
+#include <GLFW/glfw3.h>
 
 vsx_manager_abs* manager;
 
@@ -105,13 +105,13 @@
 {
   switch (key)
   {
-    case GLFW_KEY_ESC:
+    case GLFW_KEY_ESCAPE:
       if (manager) manager_destroy(manager);
       exit(0);
-    case GLFW_KEY_PAGEUP:
+    case GLFW_KEY_PAGE_UP:
       manager->inc_speed();
       break;
-    case GLFW_KEY_PAGEDOWN:
+    case GLFW_KEY_PAGE_DOWN:
       manager->dec_speed();
       break;
     case GLFW_KEY_UP:
--- vsxu-0.5.1/server/src/main/linux-opengl/main.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/server/src/main/linux-opengl/main.cpp	2016-03-02 20:21:33.300464802 +0100
@@ -27,7 +27,7 @@
 #include <vsx_string_helper.h>
 #include <vsx_argvector.h>
 #include <GL/glew.h>
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 #include "application.h"
 #include "vsxfst.h"
 #include "vsx_version.h"
@@ -67,9 +67,9 @@
 
 void set_modifiers()
 {
-  app_ctrl = (bool)glfwGetKey(GLFW_KEY_LCTRL);
-  app_shift = (bool)glfwGetKey(GLFW_KEY_LSHIFT);
-  app_alt = (bool)glfwGetKey(GLFW_KEY_LALT);
+  app_ctrl = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_CONTROL);
+  app_shift = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_SHIFT);
+  app_alt = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_ALT);
 }
 
 long key_pressed = -1;
@@ -78,18 +78,14 @@
 float key_repeat_time;
 float initial_key_delay = 0.04f;
 
-void GLFWCALL key_char_event( int character, int action )
+void key_char_event( GLFWwindow *window, unsigned int character )
 {
-  if (action == GLFW_PRESS)
-  {
     app_char(character);
     key_character = character;
-
-  }
 }
 
 
-void GLFWCALL key_event(int key, int action)
+void key_event(GLFWwindow *window, int key, int scancode, int action, int mods)
 {
   set_modifiers();
   if (action == GLFW_PRESS)
@@ -108,12 +104,12 @@
   }
 }
 
-int last_x = 0, last_y = 0;
+double last_x = 0, last_y = 0;
 int mouse_state = 0;
 
-void GLFWCALL mouse_button_event(int button, int action)
+void mouse_button_event(GLFWwindow *window, int button, int action, int mods)
 {
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   set_modifiers();
   unsigned long i_but = 0;
   switch (button)
@@ -135,26 +131,26 @@
 
 int mouse_pos_type = 0;
 
-void GLFWCALL mouse_pos_event(int x, int y)
+void mouse_pos_event(GLFWwindow *window, double x, double y)
 {
   VSX_UNUSED(x);
   VSX_UNUSED(y);
   set_modifiers();
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   if (mouse_state) mouse_pos_type = 1;
   else mouse_pos_type = 2;
 }
 
-int mousewheel_prev_pos = 0;
+double mousewheel_prev_pos = 0;
 
-void GLFWCALL mouse_wheel(int pos)
+void mouse_wheel(GLFWwindow *window, double xoffset, double yoffset)
 {
   set_modifiers();
-  app_mousewheel((float)(pos-mousewheel_prev_pos),last_x,last_y);
-  mousewheel_prev_pos = pos;
+  app_mousewheel((float)(yoffset-mousewheel_prev_pos),last_x,last_y);
+  mousewheel_prev_pos = yoffset;
 }
 
-void GLFWCALL window_size( int width, int height )
+void window_size( GLFWwindow *window, int width, int height )
 {
   vsx_gl_state::get_instance()->viewport_change(0,0,width, height);
 }
@@ -203,7 +199,8 @@
   set_modifiers();
 
 
-  int     width, height, running, frames, x, y;
+  int     width, height, running, frames;
+  double  x, y;
   double  t, t1;
 //  unsigned long usleep_framelimit;
   char    titlestr[ 200 ];
@@ -252,24 +249,24 @@
   if (start_fullscreen && !manual_resolution_set)
   {
     // try to get the resolution from the desktop for fullscreen
-    GLFWvidmode video_mode;
-    glfwGetDesktopMode(&video_mode);
-    x_res = video_mode.Height;
-    y_res = video_mode.Width;
+    GLFWvidmode video_mode = * glfwGetVideoMode(glfwGetPrimaryMonitor());
+    x_res = video_mode.height;
+    y_res = video_mode.width;
   }
 
 
   if (vsx_argvector::get_instance()->has_param("gl_debug"))
   {
     printf("enabling GL DEBUG\n");
-    glfwOpenWindowHint( GLFW_OPENGL_DEBUG_CONTEXT , GL_TRUE );
+    glfwWindowHint( GLFW_OPENGL_DEBUG_CONTEXT , GL_TRUE );
   }
 
   // OpenGL version
-  glfwOpenWindowHint(GLFW_OPENGL_VERSION_MAJOR, 2);
-  glfwOpenWindowHint(GLFW_OPENGL_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_DEPTH_BITS, 16);
 
-  if( !glfwOpenWindow( x_res, y_res, 0,0,0,0,16,0, start_fullscreen?GLFW_FULLSCREEN:GLFW_WINDOW ) ) // GLFW_FULLSCREEN
+  if( !glfwCreateWindow( x_res, y_res, "", start_fullscreen ? glfwGetPrimaryMonitor() : NULL, NULL ) )
   {
     printf("Error! Could not create an OpenGL context. Please check your GPU drivers...\n");
     glfwTerminate();
@@ -278,11 +275,9 @@
 
   glewInit();
 
-  if (start_fullscreen) glfwEnable( GLFW_MOUSE_CURSOR );
+  if (start_fullscreen) glfwSetInputMode(glfwGetCurrentContext(), GLFW_CURSOR, GLFW_CURSOR_NORMAL);
   app_init(0);
 
-  glfwEnable(GLFW_AUTO_POLL_EVENTS);
-
   for (int i = 1; i < argc; i++)
   {
     vsx_string arg1 = argv[i];
@@ -294,20 +289,20 @@
         vsx_avector<vsx_string> parts;
         vsx_string deli = ",";
         explode(arg2, deli, parts);
-        glfwSetWindowPos( vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]) );
+        glfwSetWindowPos(glfwGetCurrentContext(), vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]));
       }
     }
   }
 
-  glfwSetKeyCallback(&key_event);
-  glfwSetMouseButtonCallback(&mouse_button_event);
-  glfwSetMousePosCallback(&mouse_pos_event);
-  glfwSetCharCallback(&key_char_event);
-  glfwSetMouseWheelCallback(&mouse_wheel);
+  glfwSetKeyCallback(glfwGetCurrentContext(), &key_event);
+  glfwSetMouseButtonCallback(glfwGetCurrentContext(), &mouse_button_event);
+  glfwSetCursorPosCallback(glfwGetCurrentContext(), &mouse_pos_event);
+  glfwSetCharCallback(glfwGetCurrentContext(), &key_char_event);
+  glfwSetScrollCallback(glfwGetCurrentContext(), &mouse_wheel);
   // set window size callback function
-  glfwSetWindowSizeCallback(window_size);
+  glfwSetWindowSizeCallback(glfwGetCurrentContext(), &window_size);
   // Enable sticky keys
-  glfwEnable( GLFW_STICKY_KEYS );
+  glfwSetInputMode( glfwGetCurrentContext(), GLFW_STICKY_KEYS, GL_TRUE );
 
 
   // vsync handling
@@ -356,7 +351,7 @@
   #if PLATFORM_FAMILY == PLATFORM_FAMILY_WINDOWS
   sprintf( titlestr, "Vovoid VSXu Artiste %s [Windows %d-bit]", vsxu_ver, PLATFORM_BITS);
   #endif
-  glfwSetWindowTitle( titlestr );
+  glfwSetWindowTitle( glfwGetCurrentContext(), titlestr );
 
 
   int initial_vram_free = 0;
@@ -403,7 +398,7 @@
 
     // Get time and mouse position
     t = glfwGetTime();
-    glfwGetMousePos( &x, &y );
+    glfwGetCursorPos( glfwGetCurrentContext(), &x, &y );
     float delta = t-t1;
     t1 = t;
     if (key_pressed != -1)
@@ -428,7 +423,7 @@
     frames ++;
 
       // Get window size (may be different than the requested size)
-      glfwGetWindowSize( &width, &height );
+      glfwGetWindowSize( glfwGetCurrentContext(), &width, &height );
       height = height > 0 ? height : 1;
 
       // Set viewport
@@ -439,12 +434,12 @@
 
     app_draw(0);
 
-    glfwSwapBuffers();
+    glfwSwapBuffers(glfwGetCurrentContext());
 
 
     // Check if the ESC key was pressed or the window was closed
-    running = /*!glfwGetKey( GLFW_KEY_ESC ) &&*/
-    glfwGetWindowParam( GLFW_OPENED );
+    running = /*!glfwGetKey( glfwGetCurrentContext(), GLFW_KEY_ESCAPE ) &&*/
+    	!glfwWindowShouldClose( glfwGetCurrentContext() );
   }
 
   // Close OpenGL window and terminate GLFW
--- vsxu-0.5.1/widget/src/widgets/vsx_widget_base_edit.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/widget/src/widgets/vsx_widget_base_edit.cpp	2016-03-02 17:57:55.150826469 +0100
@@ -36,7 +36,7 @@
 #include "vsx_widget_popup_menu.h"
 #include <stdlib.h>
 #include "vsx_widget_button.h"
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 #include <gl_helper.h>
 
 vsx_widget_base_edit::vsx_widget_base_edit() {
@@ -679,7 +679,7 @@
         }
       break;
     // page up
-    case -GLFW_KEY_PAGEUP:
+    case -GLFW_KEY_PAGE_UP:
       if (!single_row) {
         for (int zz = 0; zz < characters_height*0.95; ++zz) {
         event_key_down(-GLFW_KEY_UP);
@@ -700,7 +700,7 @@
       }
       break;
     // page down
-    case -GLFW_KEY_PAGEDOWN:
+    case -GLFW_KEY_PAGE_DOWN:
       if (!single_row) {
         for (int zz = 0; zz < characters_height*0.95; ++zz) {
         event_key_down(-GLFW_KEY_DOWN,false,false,false);
@@ -757,7 +757,7 @@
       if (mirror_keystrokes_object) mirror_keystrokes_object->event_key_down(key, alt, ctrl, shift);
     break;
     // delete
-    case -GLFW_KEY_DEL:
+    case -GLFW_KEY_DELETE:
       event_key_down(-GLFW_KEY_RIGHT,false,false,false);
       event_key_down(-GLFW_KEY_BACKSPACE,false,false,false);
       process_line(carety+(int)scroll_y);
@@ -795,7 +795,7 @@
       if (mirror_keystrokes_object) mirror_keystrokes_object->event_key_down(key, alt, ctrl, shift);
     break;
     // esc
-    case -GLFW_KEY_ESC:
+    case -GLFW_KEY_ESCAPE:
     // da rest:
       if (single_row) {
         command_q_b.add_raw("cancel");
--- vsxu-0.5.1/widget/src/vsx_mouse.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/widget/src/vsx_mouse.cpp	2016-03-02 18:33:29.197403578 +0100
@@ -22,7 +22,7 @@
 */
 
 #include "vsx_mouse.h"
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 
 void vsx_mouse::set_cursor(int id)
 {
@@ -37,7 +37,7 @@
 {
   // This causes problems with knobs & arcballs, mouse visibility is disabled for now
   //glfwEnable(GLFW_MOUSE_CURSOR);
-  //glfwSetMousePos((int)(position.x), (int)(position.y));
+  //glfwSetCursorPos(glfwGetCurrentContext(), (int)(position.x), (int)(position.y));
 }
 
 void vsx_mouse::hide_cursor()
@@ -50,7 +50,7 @@
 void vsx_mouse::set_cursor_pos(float x, float y)
 {
   position.set(x,y);
-  glfwSetMousePos((int)(position.x), (int)(position.y));
+  glfwSetCursorPos(glfwGetCurrentContext(), (int)(position.x), (int)(position.y));
 }
 
 vsx_mouse::vsx_mouse()
--- vsxu-0.5.1/artiste/src/artiste_desktop.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/artiste/src/artiste_desktop.cpp	2016-03-02 22:09:12.080193754 +0100
@@ -50,7 +50,7 @@
 #include "widgets/vsx_widget_base_edit.h"
 
 #include "artiste_desktop.h"
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 
 // VSX_WIDGET_DESKTOP **************************************************************************************************
 // VSX_WIDGET_DESKTOP **************************************************************************************************
@@ -202,10 +202,10 @@
     case -GLFW_KEY_RIGHT: case -'F': case -'f':
       camera.set_movement_x( 1.0 );
     break;
-    case -GLFW_KEY_PAGEUP: case -'R': case -'r':
+    case -GLFW_KEY_PAGE_UP: case -'R': case -'r':
       camera.set_movement_z( -1.0 );
     break;
-    case -GLFW_KEY_PAGEDOWN: case -'W': case -'w':
+    case -GLFW_KEY_PAGE_DOWN: case -'W': case -'w':
       camera.set_movement_z( 1.0 );
     break;
   } // switch
@@ -235,8 +235,8 @@
       camera.set_movement_x( 0.0 );
     break;
 
-    case GLFW_KEY_PAGEUP: case 'R': case 'r':
-    case GLFW_KEY_PAGEDOWN: case 'W': case 'w':
+    case GLFW_KEY_PAGE_UP: case 'R': case 'r':
+    case GLFW_KEY_PAGE_DOWN: case 'W': case 'w':
       camera.set_movement_z( 0.0 );
     break;
 
--- vsxu-0.5.1/profiler/src/profiler_main.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/profiler/src/profiler_main.cpp	2016-03-04 16:55:06.153732911 +0100
@@ -27,7 +27,7 @@
 #include <vsx_string_helper.h>
 #include <vsx_argvector.h>
 #include <GL/glew.h>
-#include "GL/glfw.h"
+#include <GLFW/glfw3.h>
 #include "profiler_application.h"
 #include "vsxfst.h"
 #include "vsx_version.h"
@@ -69,9 +69,9 @@
 
 void set_modifiers()
 {
-  app_ctrl = (bool)glfwGetKey(GLFW_KEY_LCTRL);
-  app_shift = (bool)glfwGetKey(GLFW_KEY_LSHIFT);
-  app_alt = (bool)glfwGetKey(GLFW_KEY_LALT);
+  app_ctrl = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_CONTROL);
+  app_shift = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_SHIFT);
+  app_alt = (bool)glfwGetKey(glfwGetCurrentContext(), GLFW_KEY_LEFT_ALT);
 }
 
 long key_pressed = -1;
@@ -80,18 +80,14 @@
 float key_repeat_time;
 float initial_key_delay = 0.04f;
 
-void GLFWCALL key_char_event( int character, int action )
+void key_char_event(GLFWwindow *window, int character)
 {
-  if (action == GLFW_PRESS)
-  {
     app_char(character);
     key_character = character;
-
-  }
 }
 
 
-void GLFWCALL key_event(int key, int action)
+void key_event(GLFWwindow *window, int key, int scancode, int action, int mods)
 {
   set_modifiers();
   if (action == GLFW_PRESS)
@@ -110,12 +106,12 @@
   }
 }
 
-int last_x = 0, last_y = 0;
+double last_x = 0, last_y = 0;
 int mouse_state = 0;
 
-void GLFWCALL mouse_button_event(int button, int action)
+void mouse_button_event(GLFWwindow *window, int button, int action, int mods)
 {
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   set_modifiers();
   unsigned long i_but = 0;
   switch (button)
@@ -137,26 +133,26 @@
 
 int mouse_pos_type = 0;
 
-void GLFWCALL mouse_pos_event(int x, int y)
+void mouse_pos_event(GLFWwindow *window, double x, double y)
 {
   VSX_UNUSED(x);
   VSX_UNUSED(y);
   set_modifiers();
-  glfwGetMousePos(&last_x, &last_y);
+  glfwGetCursorPos(glfwGetCurrentContext(), &last_x, &last_y);
   if (mouse_state) mouse_pos_type = 1;
   else mouse_pos_type = 2;
 }
 
-int mousewheel_prev_pos = 0;
+double mousewheel_prev_pos = 0;
 
-void GLFWCALL mouse_wheel(int pos)
+void mouse_wheel(GLFWwindow *window, double xoffset, double yoffset)
 {
   set_modifiers();
-  app_mousewheel((float)(pos-mousewheel_prev_pos),last_x,last_y);
-  mousewheel_prev_pos = pos;
+  app_mousewheel((float)(yoffset-mousewheel_prev_pos),last_x,last_y);
+  mousewheel_prev_pos = yoffset;
 }
 
-void GLFWCALL window_size( int width, int height )
+void window_size(GLFWwindow *window, int width, int height)
 {
   vsx_gl_state::get_instance()->viewport_change(0,0,width, height);
 }
@@ -210,7 +206,8 @@
   glfwInit();
   set_modifiers();
 
-  int     width, height, running, frames, x, y;
+  int     width, height, running, frames;
+  double  x, y;
   double  t, t1;
   char    titlestr[ 200 ];
 
@@ -253,24 +250,24 @@
   if (start_fullscreen && !manual_resolution_set)
   {
     // try to get the resolution from the desktop for fullscreen
-    GLFWvidmode video_mode;
-    glfwGetDesktopMode(&video_mode);
-    x_res = video_mode.Height;
-    y_res = video_mode.Width;
+    GLFWvidmode video_mode = * glfwGetVideoMode(glfwGetPrimaryMonitor());
+    x_res = video_mode.height;
+    y_res = video_mode.width;
   }
 
 
   if (vsx_argvector::get_instance()->has_param("gl_debug"))
   {
     printf("enabling GL DEBUG\n");
-    glfwOpenWindowHint( GLFW_OPENGL_DEBUG_CONTEXT , GL_TRUE );
+    glfwWindowHint( GLFW_OPENGL_DEBUG_CONTEXT , GL_TRUE );
   }
 
   // OpenGL version
-  glfwOpenWindowHint(GLFW_OPENGL_VERSION_MAJOR, 2);
-  glfwOpenWindowHint(GLFW_OPENGL_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_DEPTH_BITS, 16);
 
-  if( !glfwOpenWindow( x_res, y_res, 0,0,0,0,16,0, start_fullscreen?GLFW_FULLSCREEN:GLFW_WINDOW ) ) // GLFW_FULLSCREEN
+  if( !glfwCreateWindow( x_res, y_res, "", start_fullscreen ? glfwGetPrimaryMonitor() : NULL, NULL ) )
   {
     printf("Error! Could not create an OpenGL context. Please check your GPU drivers...\n");
     glfwTerminate();
@@ -280,11 +277,9 @@
   glewInit();
 
 
-  if (start_fullscreen) glfwEnable( GLFW_MOUSE_CURSOR );
+  if (start_fullscreen) glfwSetInputMode(glfwGetCurrentContext(), GLFW_CURSOR, GLFW_CURSOR_NORMAL);
   app_init();
 
-  glfwEnable(GLFW_AUTO_POLL_EVENTS);
-
   for (int i = 1; i < argc; i++)
   {
     vsx_string arg1 = argv[i];
@@ -296,20 +291,20 @@
         vsx_avector<vsx_string> parts;
         vsx_string deli = ",";
         explode(arg2, deli, parts);
-        glfwSetWindowPos( vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]) );
+        glfwSetWindowPos(glfwGetCurrentContext(), vsx_string_helper::s2i(parts[0]), vsx_string_helper::s2i(parts[1]));
       }
     }
   }
 
-  glfwSetKeyCallback(&key_event);
-  glfwSetMouseButtonCallback(&mouse_button_event);
-  glfwSetMousePosCallback(&mouse_pos_event);
-  glfwSetCharCallback(&key_char_event);
-  glfwSetMouseWheelCallback(&mouse_wheel);
+  glfwSetKeyCallback(glfwGetCurrentContext(), &key_event);
+  glfwSetMouseButtonCallback(glfwGetCurrentContext(), &mouse_button_event);
+  glfwSetCursorPosCallback(glfwGetCurrentContext(), &mouse_pos_event);
+  glfwSetCharCallback(glfwGetCurrentContext(), &key_char_event);
+  glfwSetScrollCallback(glfwGetCurrentContext(), &mouse_wheel);
   // set window size callback function
-  glfwSetWindowSizeCallback(window_size);
+  glfwSetWindowSizeCallback(glfwGetCurrentContext(), &window_size);
   // Enable sticky keys
-  glfwEnable( GLFW_STICKY_KEYS );
+  glfwSetInputMode(glfwGetCurrentContext(), GLFW_STICKY_KEYS, GL_TRUE);
 
 
   // vsync handling
@@ -354,7 +349,7 @@
 
   sprintf( titlestr, "Vovoid VSXu Profiler %s [GNU/Linux %d-bit]", vsxu_ver, PLATFORM_BITS);
 
-  glfwSetWindowTitle( titlestr );
+  glfwSetWindowTitle(glfwGetCurrentContext(), titlestr);
 
 
   vsx_timer frame_delay;
@@ -376,7 +371,7 @@
 
     // Get time and mouse position
     t = glfwGetTime();
-    glfwGetMousePos( &x, &y );
+    glfwGetCursorPos(glfwGetCurrentContext(), &x, &y);
     float delta = t-t1;
     t1 = t;
     if (key_pressed != -1)
@@ -401,7 +396,7 @@
     frames ++;
 
       // Get window size (may be different than the requested size)
-      glfwGetWindowSize( &width, &height );
+      glfwGetWindowSize(glfwGetCurrentContext(), &width, &height);
       height = height > 0 ? height : 1;
 
       // Set viewport
@@ -419,7 +414,7 @@
 
     profiler->sub_begin("swapbuffers");
 
-    glfwSwapBuffers();
+    glfwSwapBuffers(glfwGetCurrentContext());
     profiler->sub_end();
 
 
@@ -440,7 +435,7 @@
 
     // Check if the ESC key was pressed or the window was closed
     running = /*!glfwGetKey( GLFW_KEY_ESC ) &&*/
-    glfwGetWindowParam( GLFW_OPENED );
+    	!glfwWindowShouldClose(glfwGetCurrentContext());
     profiler->maj_end();
 
   }
--- vsxu-0.5.1/profiler/src/profiler_desktop.cpp.orig	2014-12-14 14:01:04.000000000 +0100
+++ vsxu-0.5.1/profiler/src/profiler_desktop.cpp	2016-03-04 16:59:50.213720990 +0100
@@ -30,7 +30,7 @@
 #include "profiler_desktop.h"
 
 #include <vsx_gl_global.h>
-#include <GL/glfw.h>
+#include <GLFW/glfw3.h>
 
 void vsx_widget_desktop::init()
 {
@@ -116,10 +116,10 @@
     case -GLFW_KEY_RIGHT: case -'F': case -'f':
       camera.set_movement_x( 1.0 );
     break;
-    case -GLFW_KEY_PAGEUP: case -'R': case -'r':
+    case -GLFW_KEY_PAGE_UP: case -'R': case -'r':
       camera.set_movement_z( -1.0 );
     break;
-    case -GLFW_KEY_PAGEDOWN: case -'W': case -'w':
+    case -GLFW_KEY_PAGE_DOWN: case -'W': case -'w':
       camera.set_movement_z( 1.0 );
     break;
   } // switch
@@ -149,8 +149,8 @@
       camera.set_movement_x( 0.0 );
     break;
 
-    case GLFW_KEY_PAGEUP: case 'R': case 'r':
-    case GLFW_KEY_PAGEDOWN: case 'W': case 'w':
+    case GLFW_KEY_PAGE_UP: case 'R': case 'r':
+    case GLFW_KEY_PAGE_DOWN: case 'W': case 'w':
       camera.set_movement_z( 0.0 );
     break;
   }
